app-emulation/amazon-ssm-agent/amazon-ssm-agent-2.3.1205.0.ebuild

```diff
@@ -0,0 +1,61 @@
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+COREOS_GO_PACKAGE="${GITHUB_URI}"
+COREOS_GO_VERSION="go1.12"
+
+inherit coreos-go-depend golang-vcs-snapshot
+
+EGO_PN="github.com/aws/${PN}"
+DESCRIPTION="AWS Systems Manager Agent"
+HOMEPAGE="https://github.com/aws/amazon-ssm-agent"
+LICENSE="Apache-2.0"
+SRC_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz ${EGO_VENDOR_URI}"
+SLOT="0"
+KEYWORDS="amd64 arm64"
+
+S="${WORKDIR}/${PN}-${PV}/src/${EGO_PN}"
+
+PATCHES=(
+  # patch generated by `goimports -w ./agent` command
+  "${FILESDIR}"/${P}-goimports.patch
+)
+
+src_prepare() {
+  default
+  ln -s ${PWD}/vendor/src/*  ${PWD}/vendor/
+}
+
+src_compile() {
+  # this is replication command from the vendor makefile
+  # but without network activity during build phase
+  local GO_LDFLAGS="-s -w -extldflags=-Wl,-z,now,-z,relro,-z,defs"
+  export GOPATH="${WORKDIR}/${PN}-${PV}"
+  # set release version
+  BRAZIL_PACKAGE_VERSION=${PV} go run ./agent/version/versiongenerator/version-gen.go
```

Comment:
Hmm, what is `BRAZIL_PACKAGE_VERSION`?
I don't see it in the code of amazon-ssm-agent.

Reply:
@dongsupark this is the most crazy and funny thing i ever seen in this code (which, btw, is very bad quality). Variable is used by https://github.com/aws/amazon-ssm-agent/blob/master/agent/version/versiongenerator/version-gen.go#L60. This way they are setting version number in the releases, as [version file](https://github.com/aws/amazon-ssm-agent/blob/master/VERSION) is always showing `2.3.0.0`. I found that when i was analyzing how releases are published.

Reply:
OMG, it is indeed funny.

Why does it translate a base64-encoded string to a name for a specific env variable? It looks like a deliberately cryptic code. How could any newcomer guess the prefix `BRAZIL_`? :joy:  I have tried to find out its reason, but could find nothing from https://github.com/aws/amazon-ssm-agent/commit/57316ab84030b004eaecd19296806020a2f6a502, and am not able to log in their internal issue tracker https://cr.amazon.com/r/5572760/. Of course no docs at all about that.

Anyway thanks for the explanation.